### PR TITLE
[docs] Add AttnProcessor to docs

### PR DESCRIPTION
**Pull request by stevhliu**
Adds the `AttnProcessor` and `AttnProcessor2_0` to the API docs.

### CODE REVIEW

**Review by patrickvonplaten — COMMENTED** [review]
Great idea!

File: docs/source/en/api/attnprocessor.mdx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+## AttnProcessor2_0
+[[autodoc]] models.attention_processor.AttnProcessor2_0
+
+## LoRAAttnProcessor
+[[autodoc]] models.attention_processor.LoRAAttnProcessor
+
+## CustomDiffusionAttnProcessor
+[[autodoc]] models.attention_processor.CustomDiffusionAttnProcessor
+
+## AttnAddedKVProcessor
```
Comment (stevhliu) [inline docs/source/en/api/attnprocessor.mdx:17]: I couldn't really find anything about what `AddedKV` attention is so I'd love some help here :) cc @sayakpaul @patrickvonplaten @yiyixuxu @pcuenca

File: src/diffusers/models/attention_processor.py [hunk 2]
```diff
@@ -413,6 +413,10 @@ def norm_encoder_hidden_states(self, encoder_hidden_states):
 
 
 class AttnProcessor:
+    r"""
+    Processor for implementing the basic attention mechanism.
```
Comment (sayakpaul) [inline src/diffusers/models/attention_processor.py:417]:
```suggestion
    Default processor for performing attention-related computations.
```

File: src/diffusers/models/attention_processor.py [hunk 3]
```diff
@@ -558,6 +574,24 @@ def __call__(self, attn: Attention, hidden_states, encoder_hidden_states=None, a
 
 
 class CustomDiffusionAttnProcessor(nn.Module):
+    r"""
+    Processor for implementing attention for the Custom Diffusion method.
+
+    Args:
+        train_kv (`bool`, defaults to `True`):
+            Whether to train the text features.
```
Comment (sayakpaul) [inline src/diffusers/models/attention_processor.py:582]:
```suggestion
            Whether to newly train the key and value matrices corresponding to the text features.
```

File: src/diffusers/models/attention_processor.py [hunk 4]
```diff
@@ -558,6 +574,24 @@ def __call__(self, attn: Attention, hidden_states, encoder_hidden_states=None, a
...
+        train_q_out (`bool`, defaults to `True`):
+            Whether to train the latent image features.
```
Comment (sayakpaul) [inline src/diffusers/models/attention_processor.py:584]:
```suggestion
            Whether to newly train the query matrices corresponding to the latent features.
```

File: src/diffusers/models/attention_processor.py [hunk 5]
```diff
@@ -810,6 +854,17 @@ def __call__(self, attn: Attention, hidden_states, encoder_hidden_states=None, a
 
 
 class XFormersAttnProcessor:
+    r"""
+    Processor for implementing memory efficient attention.
```
Comment (sayakpaul) [inline src/diffusers/models/attention_processor.py:858]:
```suggestion
    Processor for implementing memory efficient attention using xformers.
```

File: src/diffusers/models/attention_processor.py [hunk 6]
```diff
@@ -936,6 +995,23 @@ def __call__(self, attn: Attention, hidden_states, encoder_hidden_states=None, a
 
 
 class LoRAXFormersAttnProcessor(nn.Module):
+    r"""
+    Processor for implementing memory efficient attention with LoRA.
```
Comment (sayakpaul) [inline src/diffusers/models/attention_processor.py:999]:
```suggestion
    Processor for implementing the LoRA attention mechanism with memory efficient attention using xformers.
```

File: docs/source/en/api/attnprocessor.mdx (hunk 1 above)
Comment (sayakpaul) [reply docs/source/en/api/attnprocessor.mdx:17]: `AttnAddedKVProcessor` is the regular `AttnProcessor` but with extra learnable matrices for keys and values. These are used on the encoder hidden states corresponding to the text encoder.

**Review by sayakpaul — APPROVED** [review]
Looking awesome!

File: src/diffusers/models/attention_processor.py [hunk 7]
```diff
@@ -868,6 +936,10 @@ def __call__(self, attn: Attention, hidden_states, encoder_hidden_states=None, a
 
 
 class AttnProcessor2_0:
+    r"""
+    Processor for implementing scaled dot product attention (enabled by default if you're using PyTorch 2.0).
```
Comment (pcuenca) [inline src/diffusers/models/attention_processor.py:940]:
```suggestion
    Processor for implementing scaled dot-product attention (enabled by default if you're using PyTorch 2.0).
```

**Review by pcuenca — APPROVED** [review]
This is great!

**Review by patrickvonplaten — APPROVED** [review]
Great docstrings! Thanks a lot!

**Pull request merged**